src/main/scala/viper/gobra/frontend/info/implementation/typing/StmtTyping.scala

```diff
@@ -43,14 +49,17 @@ trait StmtTyping extends BaseTyping { this: TypeInfoImpl =>
 
     case n@PAssignment(rights, lefts) =>
       rights.flatMap(isExpr(_).out) ++ lefts.flatMap(isExpr(_).out) ++
-        lefts.flatMap(a => assignable.errors(a)(a)) ++ multiAssignableTo.errors(rights map exprType, lefts map exprType)(n)
+        lefts.flatMap(a => assignable.errors(a)(a)) ++ multiAssignableTo.errors(rights map exprType, lefts map exprType)(n) ++
+        (lefts zip rights).flatMap(p => assignWithinBounds.errors(exprType(p._1), p._2)(n))
 
     case n@PAssignmentWithOp(right, op, left) =>
       isExpr(right).out ++ isExpr(left).out ++
         assignable.errors(left)(n) ++ compatibleWithAssOp.errors(exprType(left), op)(n) ++
-        assignableTo.errors(exprType(right), exprType(left))(n)
+        assignableTo.errors(exprType(right), exprType(left))(n) ++
+        assignWithinBounds.errors(exprType(left), right)(n)
 
     case n@PShortVarDecl(rights, lefts, _) =>
+      // TODO: add overflow checks
```

Comment:
Could it be better to add the overflow check at the arithmetic operations. For example, when you have an addition with literals and constants, then you check whether there is an overflow. That way, you do not have to do it for all assignments.

Reply:
I do not think that that would be better because:
1. you would need to add more checks for the type-checker to detect context-defined types (e.g. `var x int8 = 127 + 1` would require you to know when you are type-checking the expression `127 + 1` that it is of type `int8`. In principle, you would need to add cases for all kinds of assignments anyway.
2. if you only checked arithmetic operations, how would you check that a literal assignment overflows? e.g.
```go
var x int8 = 128 // this should overflow and the compiler treats it as an error
```

Reply:
If you have a function that gives you the type of an expressions, then you can check literals and arightmetic operations by checking that their result (if constant) is within the bounds of the expected type. Checking at assignments seems a bit problematic, since there can be other places where additions happen, for example you can have something like `v := foo(127 + 1)` where `foo` expects an int8.

Reply:
You are right, we cannot detect that case in the current state

Reply:
Do you intend to fix this in this PR or in another PR?

Reply:
If not, then please create an issue for this.

Reply:
I was just writing a fix for this :) Please take a look at this commit 5f400d7